File: c++/2.2 Algoritmos de Ordenamiento/burbuja/BurbujaSimple.cpp
#include <iostream>

using namespace std;
int main(int argc, char *argv[]) {	
	int datos[] = {8,5,3,9,1};		
	
	cout << "Arreglo original: [";
	copy(begin(datos), end(datos), ostream_iterator<int>(cout, ","));
	cout << "\b]" << endl;

	int length = sizeof(datos) / sizeof(datos[0]); //Calcula el # de elementos

	for (int recorrido = 0; recorrido < length; recorrido++) {
		for (int  indice = 0; indice < length - 1; indice++) {				
			if (datos[indice] > datos[indice + 1]) { // Intercambio
				int buffer = datos[indice];	
				datos[indice] = datos[indice + 1];
				datos[indice + 1] = buffer;
			}								
		}
	}

	cout << "Arreglo ordenado: [";
	copy(begin(datos), end(datos), ostream_iterator<int>(cout, ","));
	cout << "\b]" << endl;

	return 0;
}

File: python/2.2 Algoritmos de Ordenamiento/burbuja/BurbujaSimple.py
import numpy as np

datos = np.array([8,5,3,9,1])
print ("Arreglo original: " + str(datos))

for recorrido in range(0, datos.size, 1): # range(inicio, fin, incremento)
	for indice in range(0, datos.size - 1, 1): 				
		if (datos[indice] > datos[indice + 1]): # Intercambio
			buffer = datos[indice]
			datos[indice] = datos[indice + 1]
			datos[indice + 1] = buffer

print ("Arreglo ordenado: " + str(datos))

